nucliadb_node/src/shards/unbounded_cache_provider/async_unbounded_writer.rs

```diff
@@ -16,3 +16,155 @@
+        })
+        .await
+        .context("Blocking task panicked")??;
+        Ok(new_shard)
+    }
+
+    async fn load(&self, id: ShardId) -> NodeResult<()> {
+        let shard_path = env::shards_path_id(&id);
+
+        if self.cache.read().await.contains_key(&id) {
+            debug!("Shard {shard_path:?} is already on memory");
+            return Ok(());
+        }
+
+        // Avoid blocking while interacting with the file system
+        let _id = id.clone();
+        let shard = tokio::task::spawn_blocking(move || {
+            if !shard_path.is_dir() {
+                return Err(node_error!("Shard {shard_path:?} is not on disk"));
+            }
+            ShardWriter::open(id.clone(), &shard_path).map_err(|error| {
+                node_error!("Shard {shard_path:?} could not be loaded from disk: {error:?}")
+            })
+        })
+        .await
+        .context("Blocking task panicked")??;
+
+        self.cache.write().await.insert(_id, Arc::new(shard));
+
+        Ok(())
+    }
+
+    async fn load_all(&self) -> NodeResult<()> {
+        let shards_path = env::shards_path();
+        let mut shards = tokio::task::spawn_blocking(move || {
+            let mut shards = HashMap::new();
+            for entry in std::fs::read_dir(&shards_path)? {
+                let entry = entry?;
+                let file_name = entry.file_name().to_str().unwrap().to_string();
+                let shard_path = entry.path();
+                match ShardWriter::open(file_name.clone(), &shard_path) {
+                    Err(err) => error!("Loading shard {shard_path:?} from disk raised {err}"),
+                    Ok(shard) => {
+                        debug!("Shard loaded: {shard_path:?}");
+                        shards.insert(file_name, shard);
+                    }
+                }
+            }
+            Ok::<HashMap<String, ShardWriter>, Error>(shards)
+        })
+        .await
+        .context("Blocking task panicked")??;
+
+        {
+            let mut cache = self.cache.write().await;
+            shards.drain().for_each(|(k, v)| {
+                cache.insert(k, Arc::new(v));
+            });
+        }
+        Ok(())
+    }
+
+    async fn get(&self, id: ShardId) -> Option<Arc<ShardWriter>> {
+        self.cache.read().await.get(&id).map(Arc::clone)
+    }
+
+    async fn delete(&self, id: ShardId) -> NodeResult<()> {
+        self.cache.write().await.remove(&id);
+
+        tokio::task::spawn_blocking(move || {
+            let shard_path = env::shards_path_id(&id);
+            if shard_path.exists() {
+                debug!("Deleting shard {shard_path:?}");
+                std::fs::remove_dir_all(shard_path)?;
+            }
+            Ok(())
+        })
+        .await
+        .context("Blocking task panicked")?
+    }
+
+    async fn upgrade(&self, id: ShardId) -> NodeResult<ShardCleaned> {
+        self.cache.write().await.remove(&id);
+
+        let _id = id.clone();
```

Comment:
This style of naming variables is usually for variables that are not used but you dont want them to be dropped (like a  mutex guard).

Reply:
What do you recommend me for this case then?

Reply:
There is not really a good way of solving this in Rust without using a crate (in C++ you can control how a closure captures). So yes, `id_` is ok

Reply:
Done